src/aiodynamo/credentials.py

```diff
@@ -354,9 +355,94 @@ async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
 
 # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
 @dataclass
-class InstanceMetadataCredentials(MetadataCredentials):
+class InstanceMetadataCredentialsWithImdsV2(MetadataCredentials):
```

Comment:
suuuper nitpicky, but "InstanceMetadata" and "Imds" are redundant. I think `InstanceMetadataCredentialsV1` and `InstanceMetadataCredentialsV2` would be more appropriate

Reply:
yeah i was thinking how to name them.. `InstanceMetadataCredentialsV2`/V1 sounds good. will change